### PR TITLE
Add support for picking up `httpNodeAuth` from device yaml

### LINKED ISSUE #317: Allow HTTP security options to be set via device settings
### Description

**As a:** flow creator

**I want to:** secure http end points served by the device agent

**So that:** they are secured to the same standard as hosted-instances

---

We can easily provide the `basic auth` option as that is hardcoded into the settings file. Provided FF Team auth is trickier as it requires oauth bounce to the platform and all the bits that entails. Need to think about how to achieve that securely when the device is running outside the security boundary of the platform

### Acceptance Criteria

- Able to secure dashboard/http end points served by the device

### Requested By

- https://app-eu1.hubspot.com/contacts/26586079/record/0-1/8936601

### PR DESCRIPTION
**Pull request by Steve-Mcl**
closes #317 
closes https://github.com/FlowFuse/flowfuse/issues/4185

## Description

* Updated config loader to pick up `httpNodeAuth` from the device yaml file
* Updates existing unit tests to ensure propery is passed through to template
* Adds missing config loader unit tests 
   *  Ensures config loader handles valid/invalid values for  `httpNodeAuth`
* Addresses carrying over values entered in the provisioning yaml to the final device yaml
   * satisfies the requirement of this task and https://github.com/FlowFuse/flowfuse/issues/4185
   * suitable unit tests added for this too

### Unit tests updated/added `test/unit/lib/template-settings_spec.js`
```
  template-settings
    ✔ should load default settings
    ✔ should set the httpNodeAuth option if provided
```

### New Unit tests file `test/unit/lib/AgentManager_spec.js`
```
  Test the AgentManager
    ✔ Agent Manager should request config from FlowFuse when started in provisioning mode
```

### New Unit tests file `test/unit/lib/config_spec.js`
```
  config loader
    parseDeviceConfig
      ✔ should parse a valid device config
      httpNodeAuth
        ✔ should parse config with httpNodeAuth set false
        ✔ should parse config with valid httpNodeAuth settings
        ✔ should not validate invalid httpNodeAuth (string)
        ✔ should not validate invalid httpNodeAuth (missing pass)
    parseDeviceConfigFile
      ✔ should load and parse a valid device config file
    config
      ✔ should return a valid runtime config
      ✔ should return a valid provisioning config and any extras
      ✔ should throw for missing config file
      ✔ should throw for bad config file
```

## Related Issue(s)

#317 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [x] Documentation has been updated
    - [ ] Upgrade instructions
    - [x] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,6 +113,7 @@ Extra options   | Description
 `interval`      | How often, in seconds, the agent checks in with the platform. Default: 60s
 `intervalJitter`| How much, in seconds, to vary the heartbeat +/- `intervalJitter`. Default: 10s
 `moduleCache`   | If the device can not access npmjs.org then use the node modules cache in `module_cache` directory. Default `false`
+`httpNodeAuth`  | If set, the endpoints created in Node-RED flows will require this username and password to access them. Default: `false`
```
Comment (knolleary) [inline README.md:116]: Should this be moved to the Node-RED options table immediately below?

Comment (Steve-Mcl) [reply README.md:116]: aaagh, i fussed over where to put it and completely missed that!

Comment (Steve-Mcl) [reply README.md]: Fixed up in [improve readme for httpNodeAuth](https://github.com/FlowFuse/device-agent/pull/320/commits/2c7f80780cf07f9fc91a69b23e002b53f42740bb)

**Comment by Steve-Mcl** [comment]
Just a thought - we should probably verify this is passed through for provisioning - or is that considered out of scope?

**Comment by knolleary** [comment]
> Just a thought - we should probably verify this is passed through for provisioning - or is that considered out of scope?

Not quite sure what you mean here - can you expand?

**Review by knolleary — APPROVED** [review]
Approved, pending the check on provisioning behaviour.

**Comment by Steve-Mcl** [comment]
@knolleary I have updated this to carry over extra values in the provisioning config as per discussion in https://github.com/FlowFuse/flowfuse/issues/4185 

I added extra tests (and updated the info in the OP):
```
  Test the AgentManager
    ✔ Agent Manager should request config from FlowFuse when started in provisioning mode
```

```
  config loader
     config
      ✔ should return a valid provisioning config and any extras
```


@hardillb would appreciate your eyes for review given the above.

**Review by hardillb — APPROVED** [review]
Looks good

**Pull request merged**